// fluent-ffmpeg/lib-cov/fluent-ffmpeg.js
/* automatically generated by JSCoverage - do not edit */
try {
  if (typeof top === 'object' && top !== null && typeof top.opener === 'object' && top.opener !== null) {
    // this is a browser window that was opened from another window

    if (! top.opener._$jscoverage) {
      top.opener._$jscoverage = {};
    }
  }
}
catch (e) {}

try {
  if (typeof top === 'object' && top !== null) {
    // this is a browser window

    try {
      if (typeof top.opener === 'object' && top.opener !== null && top.opener._$jscoverage) {
        top._$jscoverage = top.opener._$jscoverage;
      }
    }
    catch (e) {}

    if (! top._$jscoverage) {
      top._$jscoverage = {};
    }
  }
}
catch (e) {}

try {
  if (typeof top === 'object' && top !== null && top._$jscoverage) {
    _$jscoverage = top._$jscoverage;
  }
}
catch (e) {}
if (typeof _$jscoverage !== 'object') {
  _$jscoverage = {};
}
if (! _$jscoverage['fluent-ffmpeg.js']) {
  _$jscoverage['fluent-ffmpeg.js'] = [];
  _$jscoverage['fluent-ffmpeg.js'][1] = 0;
  _$jscoverage['fluent-ffmpeg.js'][13] = 0;
  _$jscoverage['fluent-ffmpeg.js'][14] = 0;
  _$jscoverage['fluent-ffmpeg.js'][20] = 0;
  _$jscoverage['fluent-ffmpeg.js'][22] = 0;
  _$jscoverage['fluent-ffmpeg.js'][23] = 0;
  _$jscoverage['fluent-ffmpeg.js'][25] = 0;
  _$jscoverage['fluent-ffmpeg.js'][34] = 0;
  _$jscoverage['fluent-ffmpeg.js'][37] = 0;
  _$jscoverage['fluent-ffmpeg.js'][38] = 0;
  _$jscoverage['fluent-ffmpeg.js'][39] = 0;
  _$jscoverage['fluent-ffmpeg.js'][41] = 0;
  _$jscoverage['fluent-ffmpeg.js'][42] = 0;
  _$jscoverage['fluent-ffmpeg.js'][43] = 0;
  _$jscoverage['fluent-ffmpeg.js'][45] = 0;
  _$jscoverage['fluent-ffmpeg.js'][46] = 0;
  _$jscoverage['fluent-ffmpeg.js'][50] = 0;
  _$jscoverage['fluent-ffmpeg.js'][53] = 0;
  _$jscoverage['fluent-ffmpeg.js'][71] = 0;
  _$jscoverage['fluent-ffmpeg.js'][73] = 0;
  _$jscoverage['fluent-ffmpeg.js'][74] = 0;
  _$jscoverage['fluent-ffmpeg.js'][75] = 0;
  _$jscoverage['fluent-ffmpeg.js'][76] = 0;
  _$jscoverage['fluent-ffmpeg.js'][78] = 0;
  _$jscoverage['fluent-ffmpeg.js'][80] = 0;
  _$jscoverage['fluent-ffmpeg.js'][82] = 0;
  _$jscoverage['fluent-ffmpeg.js'][84] = 0;
  _$jscoverage['fluent-ffmpeg.js'][85] = 0;
  _$jscoverage['fluent-ffmpeg.js'][86] = 0;
  _$jscoverage['fluent-ffmpeg.js'][88] = 0;
  _$jscoverage['fluent-ffmpeg.js'][89] = 0;
  _$jscoverage['fluent-ffmpeg.js'][90] = 0;
  _$jscoverage['fluent-ffmpeg.js'][92] = 0;
  _$jscoverage['fluent-ffmpeg.js'][93] = 0;
  _$jscoverage['fluent-ffmpeg.js'][94] = 0;
  _$jscoverage['fluent-ffmpeg.js'][96] = 0;
  _$jscoverage['fluent-ffmpeg.js'][97] = 0;
  _$jscoverage['fluent-ffmpeg.js'][99] = 0;
  _$jscoverage['fluent-ffmpeg.js'][100] = 0;
  _$jscoverage['fluent-ffmpeg.js'][102] = 0;
  _$jscoverage['fluent-ffmpeg.js'][103] = 0;
  _$jscoverage['fluent-ffmpeg.js'][104] = 0;
  _$jscoverage['fluent-ffmpeg.js'][106] = 0;
  _$jscoverage['fluent-ffmpeg.js'][107] = 0;
  _$jscoverage['fluent-ffmpeg.js'][108] = 0;
  _$jscoverage['fluent-ffmpeg.js'][109] = 0;
  _$jscoverage['fluent-ffmpeg.js'][111] = 0;
  _$jscoverage['fluent-ffmpeg.js'][113] = 0;
  _$jscoverage['fluent-ffmpeg.js'][115] = 0;
  _$jscoverage['fluent-ffmpeg.js'][116] = 0;
  _$jscoverage['fluent-ffmpeg.js'][117] = 0;
  _$jscoverage['fluent-ffmpeg.js'][119] = 0;
  _$jscoverage['fluent-ffmpeg.js'][120] = 0;
  _$jscoverage['fluent-ffmpeg.js'][121] = 0;
  _$jscoverage['fluent-ffmpeg.js'][123] = 0;
  _$jscoverage['fluent-ffmpeg.js'][124] = 0;
  _$jscoverage['fluent-ffmpeg.js'][125] = 0;
  _$jscoverage['fluent-ffmpeg.js'][127] = 0;
  _$jscoverage['fluent-ffmpeg.js'][128] = 0;
  _$jscoverage['fluent-ffmpeg.js'][129] = 0;
  _$jscoverage['fluent-ffmpeg.js'][131] = 0;
  _$jscoverage['fluent-ffmpeg.js'][132] = 0;
  _$jscoverage['fluent-ffmpeg.js'][133] = 0;
  _$jscoverage['fluent-ffmpeg.js'][135] = 0;
  _$jscoverage['fluent-ffmpeg.js'][136] = 0;
  _$jscoverage['fluent-ffmpeg.js'][137] = 0;
  _$jscoverage['fluent-ffmpeg.js'][139] = 0;
  _$jscoverage['fluent-ffmpeg.js'][140] = 0;
  _$jscoverage['fluent-ffmpeg.js'][142] = 0;
  _$jscoverage['fluent-ffmpeg.js'][143] = 0;
  _$jscoverage['fluent-ffmpeg.js'][144] = 0;
  _$jscoverage['fluent-ffmpeg.js'][146] = 0;
  _$jscoverage['fluent-ffmpeg.js'][147] = 0;
  _$jscoverage['fluent-ffmpeg.js'][148] = 0;
  _$jscoverage['fluent-ffmpeg.js'][150] = 0;
  _$jscoverage['fluent-ffmpeg.js'][151] = 0;
  _$jscoverage['fluent-ffmpeg.js'][152] = 0;
  _$jscoverage['fluent-ffmpeg.js'][154] = 0;
  _$jscoverage['fluent-ffmpeg.js'][155] = 0;
  _$jscoverage['fluent-ffmpeg.js'][156] = 0;
  _$jscoverage['fluent-ffmpeg.js'][158] = 0;
  _$jscoverage['fluent-ffmpeg.js'][159] = 0;
  _$jscoverage['fluent-ffmpeg.js'][160] = 0;
  _$jscoverage['fluent-ffmpeg.js'][162] = 0;
  _$jscoverage['fluent-ffmpeg.js'][163] = 0;
  _$jscoverage['fluent-ffmpeg.js'][164] = 0;
  _$jscoverage['fluent-ffmpeg.js'][166] = 0;
  _$jscoverage['fluent-ffmpeg.js'][167] = 0;
  _$jscoverage['fluent-ffmpeg.js'][168] = 0;
  _$jscoverage['fluent-ffmpeg.js'][170] = 0;
  _$jscoverage['fluent-ffmpeg.js'][171] = 0;
  _$jscoverage['fluent-ffmpeg.js'][172] = 0;
  _$jscoverage['fluent-ffmpeg.js'][173] = 0;
  _$jscoverage['fluent-ffmpeg.js'][174] = 0;
  _$jscoverage['fluent-ffmpeg.js'][175] = 0;
  _$jscoverage['fluent-ffmpeg.js'][176] = 0;
  _$jscoverage['fluent-ffmpeg.js'][178] = 0;
  _$jscoverage['fluent-ffmpeg.js'][182] = 0;
  _$jscoverage['fluent-ffmpeg.js'][184] = 0;
  _$jscoverage['fluent-ffmpeg.js'][185] = 0;
  _$jscoverage['fluent-ffmpeg.js'][186] = 0;
  _$jscoverage['fluent-ffmpeg.js'][188] = 0;
  _$jscoverage['fluent-ffmpeg.js'][189] = 0;
  _$jscoverage['fluent-ffmpeg.js'][192] = 0;
  _$jscoverage['fluent-ffmpeg.js'][193] = 0;
  _$jscoverage['fluent-ffmpeg.js'][195] = 0;
  _$jscoverage['fluent-ffmpeg.js'][197] = 0;
  _$jscoverage['fluent-ffmpeg.js'][198] = 0;
  _$jscoverage['fluent-ffmpeg.js'][199] = 0;
  _$jscoverage['fluent-ffmpeg.js'][201] = 0;
  _$jscoverage['fluent-ffmpeg.js'][202] = 0;
  _$jscoverage['fluent-ffmpeg.js'][204] = 0;
  _$jscoverage['fluent-ffmpeg.js'][208] = 0;
  _$jscoverage['fluent-ffmpeg.js'][209] = 0;
  _$jscoverage['fluent-ffmpeg.js'][210] = 0;
  _$jscoverage['fluent-ffmpeg.js'][212] = 0;
  _$jscoverage['fluent-ffmpeg.js'][213] = 0;
  _$jscoverage['fluent-ffmpeg.js'][215] = 0;
  _$jscoverage['fluent-ffmpeg.js'][216] = 0;
  _$jscoverage['fluent-ffmpeg.js'][217] = 0;
  _$jscoverage['fluent-ffmpeg.js'][219] = 0;
  _$jscoverage['fluent-ffmpeg.js'][220] = 0;
  _$jscoverage['fluent-ffmpeg.js'][221] = 0;
  _$jscoverage['fluent-ffmpeg.js'][223] = 0;
  _$jscoverage['fluent-ffmpeg.js'][224] = 0;
  _$jscoverage['fluent-ffmpeg.js'][225] = 0;
  _$jscoverage['fluent-ffmpeg.js'][227] = 0;
  _$jscoverage['fluent-ffmpeg.js'][228] = 0;
  _$jscoverage['fluent-ffmpeg.js'][229] = 0;
  _$jscoverage['fluent-ffmpeg.js'][230] = 0;
  _$jscoverage['fluent-ffmpeg.js'][232] = 0;
  _$jscoverage['fluent-ffmpeg.js'][236] = 0;
  _$jscoverage['fluent-ffmpeg.js'][237] = 0;
  _$jscoverage['fluent-ffmpeg.js'][240] = 0;
  _$jscoverage['fluent-ffmpeg.js'][241] = 0;
  _$jscoverage['fluent-ffmpeg.js'][242] = 0;
  _$jscoverage['fluent-ffmpeg.js'][244] = 0;
  _$jscoverage['fluent-ffmpeg.js'][246] = 0;
  _$jscoverage['fluent-ffmpeg.js'][247] = 0;
  _$jscoverage['fluent-ffmpeg.js'][248] = 0;
  _$jscoverage['fluent-ffmpeg.js'][249] = 0;
  _$jscoverage['fluent-ffmpeg.js'][252] = 0;
  _$jscoverage['fluent-ffmpeg.js'][253] = 0;
  _$jscoverage['fluent-ffmpeg.js'][254] = 0;
  _$jscoverage['fluent-ffmpeg.js'][256] = 0;
  _$jscoverage['fluent-ffmpeg.js'][259] = 0;
  _$jscoverage['fluent-ffmpeg.js'][263] = 0;
  _$jscoverage['fluent-ffmpeg.js'][264] = 0;
  _$jscoverage['fluent-ffmpeg.js'][267] = 0;
  _$jscoverage['fluent-ffmpeg.js'][268] = 0;
  _$jscoverage['fluent-ffmpeg.js'][269] = 0;
  _$jscoverage['fluent-ffmpeg.js'][270] = 0;
  _$jscoverage['fluent-ffmpeg.js'][272] = 0;
  _$jscoverage['fluent-ffmpeg.js'][273] = 0;
  _$jscoverage['fluent-ffmpeg.js'][277] = 0;
  _$jscoverage['fluent-ffmpeg.js'][278] = 0;
  _$jscoverage['fluent-ffmpeg.js'][281] = 0;
  _$jscoverage['fluent-ffmpeg.js'][282] = 0;
  _$jscoverage['fluent-ffmpeg.js'][284] = 0;
  _$jscoverage['fluent-ffmpeg.js'][287] = 0;
  _$jscoverage['fluent-ffmpeg.js'][294] = 0;
  _$jscoverage['fluent-ffmpeg.js'][295] = 0;
  _$jscoverage['fluent-ffmpeg.js'][296] = 0;
  _$jscoverage['fluent-ffmpeg.js'][297] = 0;
  _$jscoverage['fluent-ffmpeg.js'][298] = 0;
  _$jscoverage['fluent-ffmpeg.js'][301] = 0;
  _$jscoverage['fluent-ffmpeg.js'][302] = 0;
  _$jscoverage['fluent-ffmpeg.js'][306] = 0;
  _$jscoverage['fluent-ffmpeg.js'][307] = 0;
  _$jscoverage['fluent-ffmpeg.js'][309] = 0;
  _$jscoverage['fluent-ffmpeg.js'][310] = 0;
}
_$jscoverage['fluent-ffmpeg.js'].source = ["var path = require('path'),","  async = require('../support/async.min.js'),","  exec = require('child_process').exec,","  spawn = require('child_process').spawn;","","/* options object consists of the following keys:"," * - source: either a ReadableStream or the path to a file (required)"," * - timeout: timeout in seconds for all ffmpeg sub-processes (optional, defaults to 30)"," * - priority: default-priority for all ffmpeg sub-processes (optional, defaults to 0)"," * - logger: add a winston logging instance (optional, default is clumsy console logging)"," * - nolog: completely disables any logging"," */","function FfmpegCommand(args) {","  var source = args.source,","      timeout = args.timeout || 30,","      priority = args.priority || 0,","      logger = args.logger || null,","      nologging = args.nolog || false;","","  if (!logger &amp;&amp; !nologging) {","    // create new winston instance","    logger = require('winston');","  } else if (!logger &amp;&amp; nologging) {","    // create fake object to route log calls","    logger = {","      debug: function() {},","      info: function() {},","      warn: function() {},","      error: function() {}","    };","  }","","  // make sure execution is not killed on error","  logger.exitOnError = false;","","  // check if argument is a stream","  var srcstream, srcfile;","  if (typeof source === 'object') {","    if (source.readable) {","      // streaming mode","      source.pause();","      srcstream = source;","      srcfile = source.path;","    } else {","      logger.error('Source is not a ReadableStream instance');","      throw new Error('Source is not a ReadableStream instance');","    }","  } else {","    // file mode","    srcfile = source;","  }","","  this.options = {","    _isStreamable: true,","    _updateFlvMetadata: false,","    _useConstantVideoBitrate: false,","    _nice: { level: priority },","    inputfile: srcfile,","    inputstream: srcstream,","    timeout: timeout,","    video: {},","    audio: {},","    additional: [],","    otherInputs: [],","    informInputAudioCodec: null,","    informInputVideoCodec: null,","    logger: logger","  };","","  // public chaining methods","  FfmpegCommand.prototype.usingPreset = function(preset) {","    // require preset (since require() works like a singleton, multiple calls generate no overhead)","    try {     ","      var module = require('./presets/' + preset);","      if (typeof module.load === 'function') {","        module.load(this);","      }","      return this;","    } catch (err) {","      throw new Error('preset ' + preset + ' could not be loaded');","    }","    return this;","  };","  FfmpegCommand.prototype.withNoVideo = function() {","    this.options.video.skip = true;","    return this;","  };","  FfmpegCommand.prototype.withNoAudio = function() {","    this.options.audio.skip = true;","    return this;","  };","  FfmpegCommand.prototype.withVideoBitrate = function(vbitrate, type) {","    if (typeof vbitrate === 'string' &amp;&amp; vbitrate.indexOf('k') &gt; 0) {","      vbitrate = vbitrate.replace('k', '');","    }","    if (type &amp;&amp; type === exports.CONSTANT_BITRATE) {","      this.options._useConstantVideoBitrate = true;","    }","    this.options.video.bitrate = parseInt(vbitrate, 10);","    return this;","  };","  FfmpegCommand.prototype.withSize = function(sizeString) {","    this.options.video.size = sizeString;","    return this;","  };","  FfmpegCommand.prototype.applyAutopadding = function(autopad, color) {","    this.options._applyAutopad = autopad;","    if (!color) {","      this.options.video.padcolor = 'black';","    } else {","      this.options.video.padcolor = color;","    }","    return this;","  };","  FfmpegCommand.prototype.withFps = function(fps) {","    this.options.video.fps = fps;","    return this;","  };","  FfmpegCommand.prototype.withFpsInput = function(fps) {","    this.options.video.fpsInput = fps;","    return this;","  };","  FfmpegCommand.prototype.withFpsOutput = function(fps) {","    this.options.video.fpsOutput = fps;","    return this;","  };","  FfmpegCommand.prototype.withAspect = function(aspectRatio) {","    this.options.video.aspect = aspectRatio;","    return this;","  };","  FfmpegCommand.prototype.withVideoCodec = function(codec) {","    this.options.video.codec = codec;","    return this;","  };","  FfmpegCommand.prototype.withAudioBitrate = function(abitrate) {","    if (typeof abitrate === 'string' &amp;&amp; abitrate.indexOf('k') &gt; 0) {","      abitrate = abitrate.replace('k', '');","    }","    this.options.audio.bitrate = parseInt(abitrate, 10);","    return this;","  };","  FfmpegCommand.prototype.withAudioCodec = function(audiocodec){","    this.options.audio.codec = audiocodec;","    return this;","  };","  FfmpegCommand.prototype.withAudioChannels = function(audiochannels) {","    this.options.audio.channels = audiochannels;","    return this;","  };","  FfmpegCommand.prototype.withAudioFrequency = function(frequency) {","    this.options.audio.frequency = frequency;","    return this;","  };","  FfmpegCommand.prototype.withAudioQuality = function(quality) {","    this.options.audio.quality = parseInt(quality, 10);","    return this;","  };","  FfmpegCommand.prototype.setStartTime = function(timestamp) {","    this.options.starttime = timestamp;","    return this;","  };","  FfmpegCommand.prototype.setDuration = function(duration) {","    this.options.duration = duration;","    return this;","  };","  FfmpegCommand.prototype.addInput = function(inputFile) {","    this.options.otherInputs.push(inputFile);","    return this;","  };","  FfmpegCommand.prototype.addOptions = function(optionArray) {","    if (typeof optionArray.length !== undefined) {","        var self = this;","        optionArray.forEach(function(el) {","          if (el.indexOf(' ') &gt; 0) {","            var values = el.split(' ');","            self.options.additional.push(values[0], values[1]);","          } else {","            self.options.additional.push(el);","          }","        });","    }","    return this;","  };","  FfmpegCommand.prototype.addOption = function(option, value) {","    this.options.additional.push(option, value);","    return this;","  };","  FfmpegCommand.prototype.toFormat = function(format) {","    this.options.format = format;","","    // some muxers require the output stream to be seekable, disable streaming for those formats","    if (this.options.format === 'mp4') {","      this.options._isStreamable = false;","    }","    return this;","  };","  FfmpegCommand.prototype.updateFlvMetadata = function() {","    this.options._updateFlvMetadata = true;","    return this;","  };","  FfmpegCommand.prototype.renice = function(level) {","    if (!level) {","      // use 0 as default nice level (os default)","      level = 0;","    }","","    // make sure niceness is within allowed boundaries","    if (level &gt; 20 || level &lt; -20) {","      this.options.logger.warn('niceness ' + level + ' is not valid, consider a value between -20 and +20 (whereas -20 is the highest priority)');","      level = 0;","    }","    this.options._nice.level = level;","    return this;","  };","  FfmpegCommand.prototype.onCodecData = function(callback) {","    this.options.onCodecData = callback;","    return this;","  };","  FfmpegCommand.prototype.onProgress = function(callback) {","    this.options.onProgress = callback;","    return this;","  };","  FfmpegCommand.prototype.gcd = function(a, b) {","    if (!a &amp;&amp; !b) {","      return 0;","    }","    while (a !== 0) {","      var z = b % a;","      b = a;","      a = z;","    }","    return b;","  };","","  // private methods","  FfmpegCommand.prototype._prepare = function(callback) {","    var calcDimensions = false, calcPadding = false;","","    // check for allowed sizestring formats and handle them accordingly","    var fixedWidth = /([0-9]+)x\\?/.exec(this.options.video.size);","    var fixedHeight = /\\?x([0-9]+)/.exec(this.options.video.size);","    var percentRatio = /\\b([0-9]{1,2})%/.exec(this.options.video.size);","","    if (!fixedWidth &amp;&amp; !fixedHeight &amp;&amp; !percentRatio) {","      // check for invalid size string","      var defaultSizestring = /([0-9]+)x([0-9]+)/.exec(this.options.video.size);","      if (this.options.video.size &amp;&amp; !defaultSizestring) {","        callback(new Error('could not parse size string, aborting execution'));","        return;","      } else {","        // get width and height as integers (used for padding calculation)","        if (defaultSizestring) {","          this.options.video.width = parseInt(defaultSizestring[1], 10);","          this.options.video.height = parseInt(defaultSizestring[2], 10);","        }","        calcDimensions = false;","      }","    } else {","      calcDimensions = true;","    }","","    // check if we have to check aspect ratio for changes and auto-pad the output","    if (this.options._applyAutopad) {","      calcPadding = true;","    }","","    var self = this;","    this.getMetadata(this.options.inputfile, function(meta, err) {","      if (calcDimensions || calcPadding) {","        var dimErr, padErr;","        // calculate dimensions","        if (calcDimensions) {","          dimErr = self._calculateDimensions(meta);","        }","","        // calculate padding","        if (calcPadding) {","          padErr = self._calculatePadding(meta);","        }","","        if (dimErr || padErr) {","          callback(new Error('error while preparing: dimension -&gt; ' + dimErr + ' padding -&gt; ' + padErr));","        } else {","          callback(undefined, meta);","        }","      } else {","        callback(undefined, meta);","      }","    });","  };","}","","// add module methods","require('./extensions').apply(FfmpegCommand.prototype);","require('./metadata').apply(FfmpegCommand.prototype);","require('./processor').apply(FfmpegCommand.prototype);","require('./calculate').apply(FfmpegCommand.prototype);","require('./debug').apply(FfmpegCommand.prototype);","","// module exports","exports = module.exports = function(args) {","  return new FfmpegCommand(args);","};","","// export meta data discovery","exports.Metadata = require('./metadata');","exports.Calculate = require('./calculate');","","exports.CONSTANT_BITRATE = 1;","exports.VARIABLE_BITRATE = 2;"];
_$jscoverage['fluent-ffmpeg.js'][1]++;
var path = require("path"), async = require("../support/async.min.js"), exec = require("child_process").exec, spawn = require("child_process").spawn;
_$jscoverage['fluent-ffmpeg.js'][13]++;
function FfmpegCommand(args) {
  _$jscoverage['fluent-ffmpeg.js'][14]++;
  var source = args.source, timeout = (args.timeout || 30), priority = (args.priority || 0), logger = (args.logger || null), nologging = (args.nolog || false);
  _$jscoverage['fluent-ffmpeg.js'][20]++;
  if (((! logger) && (! nologging))) {
    _$jscoverage['fluent-ffmpeg.js'][22]++;
    logger = require("winston");
  }
  else {
    _$jscoverage['fluent-ffmpeg.js'][23]++;
    if (((! logger) && nologging)) {
      _$jscoverage['fluent-ffmpeg.js'][25]++;
      logger = {debug: (function () {
}), info: (function () {
}), warn: (function () {
}), error: (function () {
})};
    }
  }
  _$jscoverage['fluent-ffmpeg.js'][34]++;
  logger.exitOnError = false;
  _$jscoverage['fluent-ffmpeg.js'][37]++;
  var srcstream, srcfile;
  _$jscoverage['fluent-ffmpeg.js'][38]++;
  if (((typeof source) === "object")) {
    _$jscoverage['fluent-ffmpeg.js'][39]++;
    if (source.readable) {
      _$jscoverage['fluent-ffmpeg.js'][41]++;
      source.pause();
      _$jscoverage['fluent-ffmpeg.js'][42]++;
      srcstream = source;
      _$jscoverage['fluent-ffmpeg.js'][43]++;
      srcfile = source.path;
    }
    else {
      _$jscoverage['fluent-ffmpeg.js'][45]++;
      logger.error("Source is not a ReadableStream instance");
      _$jscoverage['fluent-ffmpeg.js'][46]++;
      throw new Error("Source is not a ReadableStream instance");
    }
  }
  else {
    _$jscoverage['fluent-ffmpeg.js'][50]++;
    srcfile = source;
  }
  _$jscoverage['fluent-ffmpeg.js'][53]++;
  this.options = {_isStreamable: true, _updateFlvMetadata: false, _useConstantVideoBitrate: false, _nice: {level: priority}, inputfile: srcfile, inputstream: srcstream, timeout: timeout, video: {}, audio: {}, additional: [], otherInputs: [], informInputAudioCodec: null, informInputVideoCodec: null, logger: logger};
  _$jscoverage['fluent-ffmpeg.js'][71]++;
  FfmpegCommand.prototype.usingPreset = (function (preset) {
  _$jscoverage['fluent-ffmpeg.js'][73]++;
  try {
    _$jscoverage['fluent-ffmpeg.js'][74]++;
    var module = require(("./presets/" + preset));
    _$jscoverage['fluent-ffmpeg.js'][75]++;
    if (((typeof module.load) === "function")) {
      _$jscoverage['fluent-ffmpeg.js'][76]++;
      module.load(this);
    }
    _$jscoverage['fluent-ffmpeg.js'][78]++;
    return this;
  }
  catch (err) {
    _$jscoverage['fluent-ffmpeg.js'][80]++;
    throw new Error(("preset " + preset + " could not be loaded"));
  }
  _$jscoverage['fluent-ffmpeg.js'][82]++;
  return this;
});
  _$jscoverage['fluent-ffmpeg.js'][84]++;
  FfmpegCommand.prototype.withNoVideo = (function () {
  _$jscoverage['fluent-ffmpeg.js'][85]++;
  this.options.video.skip = true;
  _$jscoverage['fluent-ffmpeg.js'][86]++;
  return this;
});
  _$jscoverage['fluent-ffmpeg.js'][88]++;
  FfmpegCommand.prototype.withNoAudio = (function () {
  _$jscoverage['fluent-ffmpeg.js'][89]++;
  this.options.audio.skip = true;
  _$jscoverage['fluent-ffmpeg.js'][90]++;
  return this;
});
  _$jscoverage['fluent-ffmpeg.js'][92]++;
  FfmpegCommand.prototype.withVideoBitrate = (function (vbitrate, type) {
  _$jscoverage['fluent-ffmpeg.js'][93]++;
  if ((((typeof vbitrate) === "string") && (vbitrate.indexOf("k") > 0))) {
    _$jscoverage['fluent-ffmpeg.js'][94]++;
    vbitrate = vbitrate.replace("k", "");
  }
  _$jscoverage['fluent-ffmpeg.js'][96]++;
  if ((type && (type === exports.CONSTANT_BITRATE))) {
    _$jscoverage['fluent-ffmpeg.js'][97]++;
    this.options._useConstantVideoBitrate = true;
  }
  _$jscoverage['fluent-ffmpeg.js'][99]++;
  this.options.video.bitrate = parseInt(vbitrate, 10);
  _$jscoverage['fluent-ffmpeg.js'][100]++;
  return this;
});
  _$jscoverage['fluent-ffmpeg.js'][102]++;
  FfmpegCommand.prototype.withSize = (function (sizeString) {
  _$jscoverage['fluent-ffmpeg.js'][103]++;
  this.options.video.size = sizeString;
  _$jscoverage['fluent-ffmpeg.js'][104]++;
  return this;
});
  _$jscoverage['fluent-ffmpeg.js'][106]++;
  FfmpegCommand.prototype.applyAutopadding = (function (autopad, color) {
  _$jscoverage['fluent-ffmpeg.js'][107]++;
  this.options._applyAutopad = autopad;
  _$jscoverage['fluent-ffmpeg.js'][108]++;
  if ((! color)) {
    _$jscoverage['fluent-ffmpeg.js'][109]++;
    this.options.video.padcolor = "black";
  }
  else {
    _$jscoverage['fluent-ffmpeg.js'][111]++;
    this.options.video.padcolor = color;
  }
  _$jscoverage['fluent-ffmpeg.js'][113]++;
  return this;
});
  _$jscoverage['fluent-ffmpeg.js'][115]++;
  FfmpegCommand.prototype.withFps = (function (fps) {
  _$jscoverage['fluent-ffmpeg.js'][116]++;
  this.options.video.fps = fps;
  _$jscoverage['fluent-ffmpeg.js'][117]++;
  return this;
});
  _$jscoverage['fluent-ffmpeg.js'][119]++;
  FfmpegCommand.prototype.withFpsInput = (function (fps) {
  _$jscoverage['fluent-ffmpeg.js'][120]++;
  this.options.video.fpsInput = fps;
  _$jscoverage['fluent-ffmpeg.js'][121]++;
  return this;
});
  _$jscoverage['fluent-ffmpeg.js'][123]++;
  FfmpegCommand.prototype.withFpsOutput = (function (fps) {
  _$jscoverage['fluent-ffmpeg.js'][124]++;
  this.options.video.fpsOutput = fps;
  _$jscoverage['fluent-ffmpeg.js'][125]++;
  return this;
});
  _$jscoverage['fluent-ffmpeg.js'][127]++;
  FfmpegCommand.prototype.withAspect = (function (aspectRatio) {
  _$jscoverage['fluent-ffmpeg.js'][128]++;
  this.options.video.aspect = aspectRatio;
  _$jscoverage['fluent-ffmpeg.js'][129]++;
  return this;
});
  _$jscoverage['fluent-ffmpeg.js'][131]++;
  FfmpegCommand.prototype.withVideoCodec = (function (codec) {
  _$jscoverage['fluent-ffmpeg.js'][132]++;
  this.options.video.codec = codec;
  _$jscoverage['fluent-ffmpeg.js'][133]++;
  return this;
});
  _$jscoverage['fluent-ffmpeg.js'][135]++;
  FfmpegCommand.prototype.withAudioBitrate = (function (abitrate) {
  _$jscoverage['fluent-ffmpeg.js'][136]++;
  if ((((typeof abitrate) === "string") && (abitrate.indexOf("k") > 0))) {
    _$jscoverage['fluent-ffmpeg.js'][137]++;
    abitrate = abitrate.replace("k", "");
  }
  _$jscoverage['fluent-ffmpeg.js'][139]++;
  this.options.audio.bitrate = parseInt(abitrate, 10);
  _$jscoverage['fluent-ffmpeg.js'][140]++;
  return this;
});
  _$jscoverage['fluent-ffmpeg.js'][142]++;
  FfmpegCommand.prototype.withAudioCodec = (function (audiocodec) {
  _$jscoverage['fluent-ffmpeg.js'][143]++;
  this.options.audio.codec = audiocodec;
  _$jscoverage['fluent-ffmpeg.js'][144]++;
  return this;
});
  _$jscoverage['fluent-ffmpeg.js'][146]++;
  FfmpegCommand.prototype.withAudioChannels = (function (audiochannels) {
  _$jscoverage['fluent-ffmpeg.js'][147]++;
  this.options.audio.channels = audiochannels;
  _$jscoverage['fluent-ffmpeg.js'][148]++;
  return this;
});
  _$jscoverage['fluent-ffmpeg.js'][150]++;
  FfmpegCommand.prototype.withAudioFrequency = (function (frequency) {
  _$jscoverage['fluent-ffmpeg.js'][151]++;
  this.options.audio.frequency = frequency;
  _$jscoverage['fluent-ffmpeg.js'][152]++;
  return this;
});
  _$jscoverage['fluent-ffmpeg.js'][154]++;
  FfmpegCommand.prototype.withAudioQuality = (function (quality) {
  _$jscoverage['fluent-ffmpeg.js'][155]++;
  this.options.audio.quality = parseInt(quality, 10);
  _$jscoverage['fluent-ffmpeg.js'][156]++;
  return this;
});
  _$jscoverage['fluent-ffmpeg.js'][158]++;
  FfmpegCommand.prototype.setStartTime = (function (timestamp) {
  _$jscoverage['fluent-ffmpeg.js'][159]++;
  this.options.starttime = timestamp;
  _$jscoverage['fluent-ffmpeg.js'][160]++;
  return this;
});
  _$jscoverage['fluent-ffmpeg.js'][162]++;
  FfmpegCommand.prototype.setDuration = (function (duration) {
  _$jscoverage['fluent-ffmpeg.js'][163]++;
  this.options.duration = duration;
  _$jscoverage['fluent-ffmpeg.js'][164]++;
  return this;
});
  _$jscoverage['fluent-ffmpeg.js'][166]++;
  FfmpegCommand.prototype.addInput = (function (inputFile) {
  _$jscoverage['fluent-ffmpeg.js'][167]++;
  this.options.otherInputs.push(inputFile);
  _$jscoverage['fluent-ffmpeg.js'][168]++;
  return this;
});
  _$jscoverage['fluent-ffmpeg.js'][170]++;
  FfmpegCommand.prototype.addOptions = (function (optionArray) {
  _$jscoverage['fluent-ffmpeg.js'][171]++;
  if (((typeof optionArray.length) !== undefined)) {
    _$jscoverage['fluent-ffmpeg.js'][172]++;
    var self = this;
    _$jscoverage['fluent-ffmpeg.js'][173]++;
    optionArray.forEach((function (el) {
  _$jscoverage['fluent-ffmpeg.js'][174]++;
  if ((el.indexOf(" ") > 0)) {
    _$jscoverage['fluent-ffmpeg.js'][175]++;
    var values = el.split(" ");
    _$jscoverage['fluent-ffmpeg.js'][176]++;
    self.options.additional.push(values[0], values[1]);
  }
  else {
    _$jscoverage['fluent-ffmpeg.js'][178]++;
    self.options.additional.push(el);
  }
}));
  }
  _$jscoverage['fluent-ffmpeg.js'][182]++;
  return this;
});
  _$jscoverage['fluent-ffmpeg.js'][184]++;
  FfmpegCommand.prototype.addOption = (function (option, value) {
  _$jscoverage['fluent-ffmpeg.js'][185]++;
  this.options.additional.push(option, value);
  _$jscoverage['fluent-ffmpeg.js'][186]++;
  return this;
});
  _$jscoverage['fluent-ffmpeg.js'][188]++;
  FfmpegCommand.prototype.toFormat = (function (format) {
  _$jscoverage['fluent-ffmpeg.js'][189]++;
  this.options.format = format;
  _$jscoverage['fluent-ffmpeg.js'][192]++;
  if ((this.options.format === "mp4")) {
    _$jscoverage['fluent-ffmpeg.js'][193]++;
    this.options._isStreamable = false;
  }
  _$jscoverage['fluent-ffmpeg.js'][195]++;
  return this;
});
  _$jscoverage['fluent-ffmpeg.js'][197]++;
  FfmpegCommand.prototype.updateFlvMetadata = (function () {
  _$jscoverage['fluent-ffmpeg.js'][198]++;
  this.options._updateFlvMetadata = true;
  _$jscoverage['fluent-ffmpeg.js'][199]++;
  return this;
});
  _$jscoverage['fluent-ffmpeg.js'][201]++;
  FfmpegCommand.prototype.renice = (function (level) {
  _$jscoverage['fluent-ffmpeg.js'][202]++;
  if ((! level)) {
    _$jscoverage['fluent-ffmpeg.js'][204]++;
    level = 0;
  }
  _$jscoverage['fluent-ffmpeg.js'][208]++;
  if (((level > 20) || (level < -20))) {
    _$jscoverage['fluent-ffmpeg.js'][209]++;
    this.options.logger.warn(("niceness " + level + " is not valid, consider a value between -20 and +20 (whereas -20 is the highest priority)"));
    _$jscoverage['fluent-ffmpeg.js'][210]++;
    level = 0;
  }
  _$jscoverage['fluent-ffmpeg.js'][212]++;
  this.options._nice.level = level;
  _$jscoverage['fluent-ffmpeg.js'][213]++;
  return this;
});
  _$jscoverage['fluent-ffmpeg.js'][215]++;
  FfmpegCommand.prototype.onCodecData = (function (callback) {
  _$jscoverage['fluent-ffmpeg.js'][216]++;
  this.options.onCodecData = callback;
  _$jscoverage['fluent-ffmpeg.js'][217]++;
  return this;
});
  _$jscoverage['fluent-ffmpeg.js'][219]++;
  FfmpegCommand.prototype.onProgress = (function (callback) {
  _$jscoverage['fluent-ffmpeg.js'][220]++;
  this.options.onProgress = callback;
  _$jscoverage['fluent-ffmpeg.js'][221]++;
  return this;
});
  _$jscoverage['fluent-ffmpeg.js'][223]++;
  FfmpegCommand.prototype.gcd = (function (a, b) {
  _$jscoverage['fluent-ffmpeg.js'][224]++;
  if (((! a) && (! b))) {
    _$jscoverage['fluent-ffmpeg.js'][225]++;
    return 0;
  }
  _$jscoverage['fluent-ffmpeg.js'][227]++;
  while ((a !== 0)) {
    _$jscoverage['fluent-ffmpeg.js'][228]++;
    var z = (b % a);
    _$jscoverage['fluent-ffmpeg.js'][229]++;
    b = a;
    _$jscoverage['fluent-ffmpeg.js'][230]++;
    a = z;
}
  _$jscoverage['fluent-ffmpeg.js'][232]++;
  return b;
});
  _$jscoverage['fluent-ffmpeg.js'][236]++;
  FfmpegCommand.prototype._prepare = (function (callback) {
  _$jscoverage['fluent-ffmpeg.js'][237]++;
  var calcDimensions = false, calcPadding = false;
  _$jscoverage['fluent-ffmpeg.js'][240]++;
  var fixedWidth = /([0-9]+)x\?/.exec(this.options.video.size);
  _$jscoverage['fluent-ffmpeg.js'][241]++;
  var fixedHeight = /\?x([0-9]+)/.exec(this.options.video.size);
  _$jscoverage['fluent-ffmpeg.js'][242]++;
  var percentRatio = /\b([0-9]{1,2})%/.exec(this.options.video.size);
  _$jscoverage['fluent-ffmpeg.js'][244]++;
  if (((! fixedWidth) && (! fixedHeight) && (! percentRatio))) {
    _$jscoverage['fluent-ffmpeg.js'][246]++;
    var defaultSizestring = /([0-9]+)x([0-9]+)/.exec(this.options.video.size);
    _$jscoverage['fluent-ffmpeg.js'][247]++;
    if ((this.options.video.size && (! defaultSizestring))) {
      _$jscoverage['fluent-ffmpeg.js'][248]++;
      callback(new Error("could not parse size string, aborting execution"));
      _$jscoverage['fluent-ffmpeg.js'][249]++;
      return;
    }
    else {
      _$jscoverage['fluent-ffmpeg.js'][252]++;
      if (defaultSizestring) {
        _$jscoverage['fluent-ffmpeg.js'][253]++;
        this.options.video.width = parseInt(defaultSizestring[1], 10);
        _$jscoverage['fluent-ffmpeg.js'][254]++;
        this.options.video.height = parseInt(defaultSizestring[2], 10);
      }
      _$jscoverage['fluent-ffmpeg.js'][256]++;
      calcDimensions = false;
    }
  }
  else {
    _$jscoverage['fluent-ffmpeg.js'][259]++;
    calcDimensions = true;
  }
  _$jscoverage['fluent-ffmpeg.js'][263]++;
  if (this.options._applyAutopad) {
    _$jscoverage['fluent-ffmpeg.js'][264]++;
    calcPadding = true;
  }
  _$jscoverage['fluent-ffmpeg.js'][267]++;
  var self = this;
  _$jscoverage['fluent-ffmpeg.js'][268]++;
  this.getMetadata(this.options.inputfile, (function (meta, err) {
  _$jscoverage['fluent-ffmpeg.js'][269]++;
  if ((calcDimensions || calcPadding)) {
    _$jscoverage['fluent-ffmpeg.js'][270]++;
    var dimErr, padErr;
    _$jscoverage['fluent-ffmpeg.js'][272]++;
    if (calcDimensions) {
      _$jscoverage['fluent-ffmpeg.js'][273]++;
      dimErr = self._calculateDimensions(meta);
    }
    _$jscoverage['fluent-ffmpeg.js'][277]++;
    if (calcPadding) {
      _$jscoverage['fluent-ffmpeg.js'][278]++;
      padErr = self._calculatePadding(meta);
    }
    _$jscoverage['fluent-ffmpeg.js'][281]++;
    if ((dimErr || padErr)) {
      _$jscoverage['fluent-ffmpeg.js'][282]++;
      callback(new Error(("error while preparing: dimension -> " + dimErr + " padding -> " + padErr)));
    }
    else {
      _$jscoverage['fluent-ffmpeg.js'][284]++;
      callback(undefined, meta);
    }
  }
  else {
    _$jscoverage['fluent-ffmpeg.js'][287]++;
    callback(undefined, meta);
  }
}));
});
}
_$jscoverage['fluent-ffmpeg.js'][294]++;
require("./extensions").apply(FfmpegCommand.prototype);
_$jscoverage['fluent-ffmpeg.js'][295]++;
require("./metadata").apply(FfmpegCommand.prototype);
_$jscoverage['fluent-ffmpeg.js'][296]++;
require("./processor").apply(FfmpegCommand.prototype);
_$jscoverage['fluent-ffmpeg.js'][297]++;
require("./calculate").apply(FfmpegCommand.prototype);
_$jscoverage['fluent-ffmpeg.js'][298]++;
require("./debug").apply(FfmpegCommand.prototype);
_$jscoverage['fluent-ffmpeg.js'][301]++;
exports = (module.exports = (function (args) {
  _$jscoverage['fluent-ffmpeg.js'][302]++;
  return new FfmpegCommand(args);
}));
_$jscoverage['fluent-ffmpeg.js'][306]++;
exports.Metadata = require("./metadata");
_$jscoverage['fluent-ffmpeg.js'][307]++;
exports.Calculate = require("./calculate");
_$jscoverage['fluent-ffmpeg.js'][309]++;
exports.CONSTANT_BITRATE = 1;
_$jscoverage['fluent-ffmpeg.js'][310]++;
exports.VARIABLE_BITRATE = 2;
